src/App.test.jsx

```diff
@@ -1,8 +1,39 @@
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
 import App from './App';
+import userEvent from '@testing-library/user-event';
 
+//component test
 test('should render a header', async () => {
   render(<App />);
   const header = await screen.findByRole('heading', { name: /us holidays/i });
   expect(header).toBeInTheDocument();
 });
+
+//behavioral test
+
+test('should display user search returning one matching item correctly', async () => {
+  render(<App />);
+  const search = await screen.findByRole('textbox');
+  const button = await screen.findByRole('button');
+  userEvent.type(search, `New Year's Day`);
+  userEvent.click(button);
+  await waitFor(() => expect(screen.getAllByRole('heading')).toHaveLength(4));
```

Comment:
Because you're applications is searching through local state this `await waitFor` may not be necessary